MariaDB can’t be launched (Supervisor error)

Run command :
```
docker run -d --name="xxxxx_DB" --volumes-from="xxxxx_DBDATA" -e "MARIADB_USER=xxxxx" -e "MARIADB_PASS=xxxxxxx" --restart=always maxexcloo/mariadb
```

MariaDB logs :
```
2016-09-26 12:09:47,104 INFO exited: mariadb (exit status 0; expected)
/usr/lib/python2.7/dist-packages/supervisor/options.py:296: UserWarning: Supervisord is running as root and it is searching for its configuration file in default locations (including its current working directory); you probably want to specify a "-c" argument specifying an absolute path to a configuration file for improved security.
  'Supervisord is running as root and it is searching '
2016-09-26 12:23:26,874 CRIT Supervisor running as root (no user in config file)
2016-09-26 12:23:26,874 WARN Included extra file "/etc/supervisor/conf.d/mariadb.conf" during parsing
Unlinking stale socket /var/run/supervisor.sock
```

```
2016-09-26 14:17:38,386 CRIT Server 'unix_http_server' running without any HTTP authentication checking
```

My docker version :
```
➜  docker -v
Docker version 1.12.1, build 23cf638
```

After some searching… executing in mariadb container:

```
mysqld_safe --skip-syslog
```

Logs give me error about permissions: 

```
160926 14:36:47 mysqld_safe Starting mysqld daemon with databases from /data
2016-09-26 14:36:47 139856825276352 [Note] /usr/sbin/mysqld (mysqld 10.1.17-MariaDB-1~jessie) starting as process 3016 ...
2016-09-26 14:36:47 139856825276352 [ERROR] mysqld: File '/var/log/mysql/mariadb-bin.index' not found (Errcode: 13 "Permission denied")
2016-09-26 14:36:47 139856825276352 [ERROR] Aborting
```

OK, I do some `chown -R core:adm /var/log/mysql` and some new errors appear.

```
160926 14:36:47 mysqld_safe mysqld from pid file /var/run/mysqld/mysqld.pid ended
160926 14:38:16 mysqld_safe Starting mysqld daemon with databases from /data
2016-09-26 14:38:16 139661845735360 [Note] /usr/sbin/mysqld (mysqld 10.1.17-MariaDB-1~jessie) starting as process 3241 ...
2016-09-26 14:38:16 139661845735360 [Note] InnoDB: Using mutexes to ref count buffer pool pages
2016-09-26 14:38:16 139661845735360 [Note] InnoDB: The InnoDB memory heap is disabled
2016-09-26 14:38:16 139661845735360 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
2016-09-26 14:38:16 139661845735360 [Note] InnoDB: GCC builtin __atomic_thread_fence() is used for memory barrier
2016-09-26 14:38:16 139661845735360 [Note] InnoDB: Compressed tables use zlib 1.2.8
2016-09-26 14:38:16 139661845735360 [Note] InnoDB: Using Linux native AIO
2016-09-26 14:38:16 139661845735360 [Note] InnoDB: Using SSE crc32 instructions
2016-09-26 14:38:16 139661845735360 [Note] InnoDB: Initializing buffer pool, size = 256.0M
2016-09-26 14:38:17 139661845735360 [Note] InnoDB: Completed initialization of buffer pool
2016-09-26 14:38:17 139661845735360 [Note] InnoDB: The first specified data file ./ibdata1 did not exist: a new database to be created!
2016-09-26 14:38:17 139661845735360 [Note] InnoDB: Setting file ./ibdata1 size to 12 MB
2016-09-26 14:38:17 139661845735360 [Note] InnoDB: Database physically writes the file full: wait...
2016-09-26 14:38:17 139661845735360 [Note] InnoDB: Setting log file ./ib_logfile101 size to 48 MB
2016-09-26 14:38:17 139661845735360 [Note] InnoDB: Setting log file ./ib_logfile1 size to 48 MB
2016-09-26 14:38:18 139661845735360 [Note] InnoDB: Renaming log file ./ib_logfile101 to ./ib_logfile0
2016-09-26 14:38:18 139661845735360 [Warning] InnoDB: New log files created, LSN=45883
2016-09-26 14:38:18 139661845735360 [Note] InnoDB: Doublewrite buffer not found: creating new
2016-09-26 14:38:18 139661845735360 [Note] InnoDB: Doublewrite buffer created
2016-09-26 14:38:18 139661845735360 [Note] InnoDB: 128 rollback segment(s) are active.
2016-09-26 14:38:18 139661845735360 [Warning] InnoDB: Creating foreign key constraint system tables.
2016-09-26 14:38:18 139661845735360 [Note] InnoDB: Foreign key constraint system tables created
2016-09-26 14:38:18 139661845735360 [Note] InnoDB: Creating tablespace and datafile system tables.
2016-09-26 14:38:18 139661845735360 [Note] InnoDB: Tablespace and datafile system tables created.
2016-09-26 14:38:18 139661845735360 [Note] InnoDB: Waiting for purge to start
2016-09-26 14:38:18 139661845735360 [Note] InnoDB:  Percona XtraDB (http://www.percona.com) 5.6.31-77.0 started; log sequence number 0
2016-09-26 14:38:18 139661049562880 [Note] InnoDB: Dumping buffer pool(s) not yet started
2016-09-26 14:38:18 139661845735360 [Note] Plugin 'FEEDBACK' is disabled.
2016-09-26 14:38:18 139661845735360 [ERROR] Could not open mysql.plugin table. Some plugins may be not loaded
2016-09-26 14:38:18 139661845494528 [Warning] Failed to load slave replication state from table mysql.gtid_slave_pos: 1146: Table 'mysql.gtid_slave_pos' doesn't exist
2016-09-26 14:38:18 139661845735360 [ERROR] Can't open and lock privilege tables: Table 'mysql.servers' doesn't exist
2016-09-26 14:38:18 139661845735360 [Note] Server socket created on IP: '::'.
2016-09-26 14:38:18 139661845735360 [ERROR] Fatal error: Can't open and lock privilege tables: Table 'mysql.user' doesn't exist
160926 14:38:18 mysqld_safe mysqld from pid file /var/run/mysqld/mysqld.pid ended
```

--- a/services/mariadb/Dockerfile
+++ b/services/mariadb/Dockerfile
@@ -1,7 +1,9 @@
 FROM maxexcloo/debian:latest
 MAINTAINER Max Schaefer <max@excloo.com>
-ENV MARIADB_USER docker \
-	MARIADB_PASS docker
+
+ENV MARIADB_USER=docker
+ENV MARIADB_PASS=docker
+
 RUN apt-key adv --keyserver keyserver.ubuntu.com --recv-keys CBCB082A1BB943DB && \
 	echo "deb http://ftp.osuosl.org/pub/mariadb/repo/10.1/debian/ jessie main" > /etc/apt/sources.list.d/mariadb.list && \
 	apt-get update && \
@@ -11,7 +13,12 @@ RUN apt-key adv --keyserver keyserver.ubuntu.com --recv-keys CBCB082A1BB943DB &&
 	rm -rf /tmp/* /var/lib/apt/lists/* /var/tmp/*
 RUN sed -i -e "s/^bind-address/#bind-address/" /etc/mysql/my.cnf && \
 	sed -i -e "s/^datadir.*=.*/datadir = \/data/" /etc/mysql/my.cnf && \
-	sed -i -e "s/^user.*=.*/user = core/" /etc/mysql/my.cnf
+	sed -i -e "s/^user.*=.*/user = core/" /etc/mysql/my.cnf && \
+	sed -i -e "s/\/var\/log\/mysql/\/data\/log/" /etc/mysql/my.cnf && \
+	chown -R core:adm /var/log/mysql.err && \
+	chown -R core:adm /var/log/mysql.log && \
+	chown -R core:adm /var/log/mysql && \
+	chown -R core:root /run/mysqld
 ADD config /config
 ADD supervisord.conf /etc/supervisor/conf.d/mariadb.conf
 EXPOSE 3306

--- a/services/mariadb/config/init/mariadb
+++ b/services/mariadb/config/init/mariadb
@@ -1,9 +1,16 @@
-if [ ! "$(ls -A /data)" ]; then
+if [ ! "$(ls -A /data/aria_log_control)" ]; then
+	mkdir -p /data/log
+	chown -R core:core /data/log
 	mysql_install_db --datadir=/data --user=core
 	mysqld_safe --skip-grant-tables &
 	while [ ! -e /run/mysqld/mysqld.sock ]; do
 		inotifywait -e create -q /run/mysqld/
 	done
-	mysql -u root -e "FLUSH PRIVILEGES; CREATE USER '$MARIADB_USER'@'%' IDENTIFIED BY '$MARIADB_PASS'; GRANT ALL PRIVILEGES ON *.* TO '$MARIADB_USER'@'%' WITH GRANT OPTION;"
+	mysql -u root -e "FLUSH PRIVILEGES; CREATE USER '${MARIADB_USER}'@'%' IDENTIFIED BY '${MARIADB_PASS}'; GRANT ALL PRIVILEGES ON *.* TO '${MARIADB_USER}'@'%' WITH GRANT OPTION;"
+	mysql -u root -e "CREATE DATABASE IF NOT EXISTS ${MARIADB_USER};"
 	mysqladmin -u root shutdown
 fi
+
+# Make socket writable for core
+chown -R core:root /run/mysqld
+chown -R core:core /data/log

